R/git.R

```diff
@@ -152,9 +152,12 @@ git_latest_commit <- function(branch = "master", root = NULL) {
 }
 
 get_reports <- function(branch, commit, show_all, default_branch, root) {
-  list_all <- show_all || identical(branch, default_branch) || is.null(branch)
+  list_all <- show_all || identical(branch, default_branch) || is_empty(branch)
   if (list_all) {
-    if (is.null(commit)) {
+    if (is_empty(commit)) {
+      if (is_empty(branch)) {
+        branch <- default_branch
+      }
       commit <- git_latest_commit(branch, root = root)
```

Comment:
Previously this would have returned latest commit on whatever branch is checked out. Now we explicitly get it for the default branch which is what the spec suggests we wanted to do